src/ensembl/src/content/app/entity-viewer/gene-view/components/transcripts-filter/TranscriptsFilter.test.tsx

```diff
@@ -0,0 +1,150 @@
+import TranscriptsFilter from './TranscriptsFilter';
+
+import RadioGroup from 'src/shared/components/radio-group/RadioGroup';
+
+const mockState = {
+  entityViewer: {
+    general: {
+      activeGenomeId: 'human',
+      activeEnsObjectIds: {
+        human: 'gene:brca2'
+      }
+    },
+    geneView: {
+      transcripts: {
+        human: {
+          'gene:brca2': {
+            expandedIds: [],
+            expandedDownloadIds: [],
+            filters: [],
+            sortingRule: 'default'
+          }
+        }
+      }
+    },
+    sidebar: {
+      human: {
+        status: Status.OPEN
+      }
+    }
+  }
+};
+
+const mockStore = configureMockStore([thunk]);
+
+const createProteinCodingTranscript = () => createTranscript();
+const createNonCodingTranscript = () => {
+  const transcript = createTranscript();
+  transcript.product_generating_contexts = [];
+  return transcript;
+};
+
+const proteinCodingTranscript1 = createProteinCodingTranscript();
+const proteinCodingTranscript2 = createProteinCodingTranscript();
+const nonCodingTranscript1 = createNonCodingTranscript();
+const nonCodingTranscript2 = createNonCodingTranscript();
+
+const defaultTranscripts = [
+  proteinCodingTranscript1,
+  proteinCodingTranscript2,
+  nonCodingTranscript1,
+  nonCodingTranscript2
+];
+
+const mockToggleFilter = jest.fn();
+
+const wrapInRedux = (
+  state: typeof mockState = mockState,
+  transcripts = defaultTranscripts
+) => {
+  return mount(
+    <Provider store={mockStore(state)}>
+      <TranscriptsFilter
+        transcripts={transcripts}
+        toggleFilter={mockToggleFilter}
+      />
+    </Provider>
+  );
+};
+
+describe('<TranscriptsFilter />', () => {
+  it('shows correct sorting option as selected', () => {
+    // default sorting option
+    let wrapper = wrapInRedux();
+    const defaultSortingLabel = wrapper
+      .find('label')
+      .findWhere((el) => el.text() === 'Default');
+    const defaultSortingRadioButton = defaultSortingLabel.find('input');
+    expect(defaultSortingRadioButton.prop('checked')).toBe(true);
+
+    // after we change sorting option
+    const updatedState = set(
+      'entityViewer.geneView.transcripts.human.gene:brca2.sortingRule',
+      'spliced_length_desc',
+      mockState
+    );
+    wrapper = wrapInRedux(updatedState);
+
+    const secondSortingLabel = wrapper
+      .find('label')
+      .findWhere(
+        (el) => el.text() === 'Combined exon length: longest – shortest'
+      );
+    const secondSortingRadioButton = secondSortingLabel.find('input');
+
+    expect(secondSortingRadioButton.prop('checked')).toBe(true);
+  });
+
+  it('correctly handles sorting order change', () => {
+    const store = mockStore(mockState);
+    const wrapper = mount(
+      <Provider store={store}>
+        <TranscriptsFilter
+          transcripts={defaultTranscripts}
+          toggleFilter={mockToggleFilter}
+        />
+      </Provider>
+    );
```

Comment:
const wrapper = wrapInRedux(); ?

Reply:
there was a reason why we couldn't use wrapInRedux but not sure what it was. I know we tried to use wrapInRedux and it complains about something. Maybe @azangru can explain.

Reply:
I think it is done this way to access the store.